slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -218,6 +223,69 @@ std::string SLAMServiceImpl::PaintMap() {
     return jpeg_img;
 }
 
+bool SLAMServiceImpl::ExtractPointCloudToBuffer(std::stringbuf &buffer) {
+    std::ostream oss(&buffer);
+    bool has_points = false;
+
+    cartographer::mapping::MapById<cartographer::mapping::NodeId,
+                                   cartographer::mapping::TrajectoryNode>
+        trajectory_nodes;
+    {
+        std::lock_guard<std::mutex> lk(map_builder_mutex);
+        trajectory_nodes =
+            map_builder.map_builder_->pose_graph()->GetTrajectoryNodes();
```

Comment:
Once we extract `trajectory_nodes` is it possible for them to be modified by `map_builder`? I want to check that accesses to `trajectory_nodes` are safe after we release the mutex.

Reply:
Looked into it together, and looks to be a safe copy including one shared pointer